api/src/main/java/com/cloud/storage/Snapshot.java

```diff
@@ -48,7 +48,7 @@ public boolean equals(String snapshotType) {
     }
 
     public enum State {
-        Allocated, Creating, CreatedOnPrimary, BackingUp, BackedUp, Copying, Destroying, Destroyed,
+        Allocated, AllocatedKVM, Creating, CreatingForVM, CreatedOnPrimary, CreatedOnPrimaryForVM, BackingUp, BackingUpForVM, BackedUp, BackedUpForVM, Copying, Destroying, Destroyed,
```

Comment:
@slavkap Snapshot states should be generic, irrespective of hypervisor. Any other thoughts?

Reply:
Hello @sureshanaparti  , thanks for the review! The design of the current implementation was to change as less as possible in CloudStack, that's why we are reusing a lot of the functionality. The additional snapshot's states are added to distinguish with the existing functionality of VM snapshots. In this PR we are lying to CloudStack that a VM snapshot is created, actually we create a separate snapshot for each disk of the VM (and they are consistent, because of the freeze of VM). And the additional states are needed to forbid the user at the end to delete one of the volume's snapshots (by mistake or on purpose). 
I'd be glad if you can give me an advice with another option

Reply:
@sureshanaparti @slavkap changing from _AllocatedKVM_ to _AllocatedVM_ would be better indeed. What do you think @sureshanaparti?

Reply:
@GabrielBrascher I've renamed it to AllocatedVM, @sureshanaparti is it OK to change only the name or I have to think to implement it in different way?